src/circularbuffer.jl

```diff
@@ -0,0 +1,61 @@
+
+"""
+New items are pushed to the back of the list, overwriting values in a circular fashion.
+"""
+type CircularBuffer{T} <: AbstractVector{T}
+    capacity::Int
+    first::Int
+    buffer::Vector{T}
+
+    CircularBuffer(capacity::Int) = new(capacity, 1, T[])
+end
+
+function _buffer_index(cb::CircularBuffer, i::Int)
+    n = length(cb)
+    if i < 1 || i > n
+        error("CircularBuffer out of range. cb=$cb i=$i")
```

Comment:
I agree, but what is the right way to support 0.3 with a BoundsError?  The signatures are different 0.3 vs 0.4.